Fixed death when landing on Slime Blocks

dktapps opened the pull request:
This commit fixes players being killed by fall damage when bouncing on Slime Blocks.

~~Also removed some broken code that was causing players to fall too fast when bouncing.~~
EDIT: see below

ishitatsuyuki commented on src/pocketmine/Player.php:
```diff
@@ -1924,12 +1924,9 @@ public function onUpdate($currentTick){
+							if(!$this->hasEffect(Effect::JUMP) and !$this->server->getAllowFlight()){
 								/*if($this->inAirTicks < 1000){
-
+									
```
Correct this formatting.

ishitatsuyuki commented:
Some formatting issues and I don't think what you are removing is problematic. -1.

dktapps commented:
The code I removed from Player.php was causing players to fall too fast when bouncing on slime. I'll fix the formatting

dktapps commented:
I've reverted the changes I made to Player.php as I realized that they broke anti-fly completely. I've added comments in instead to note that:
```
$expectedVelocity = (-$this->gravity) / $this->drag - ((-$this->gravity) / $this->drag) * exp(-$this->drag * ($this->inAirTicks - $this->startAirTicks));
```
calculates the fall velocity incorrectly, which causes issues when bouncing when antiFly is turned on.

ishitatsuyuki commented:
Oh? I think bouncing is making a up velocity, and it should work.

dktapps commented:
You'll need my patch for Entity.php to try this without getting killed, but try this
(if you want to try this without merging, I have a phar you can use to test on my [Google Drive](https://drive.google.com/open?id=0By8Xy-T6X5Z8VzNFc18ydGN5QlU) )

1. Make sure `anti-fly` is set to `false` in the `genisys.yml`.
2. Fall on slime from ~20 blocks up or so in survival. Watch how high you bounce when you fall, and how fast you fall.
3. Then turn`anti-fly` to `true` and restart the server
4. Repeat step 2. After the initial drop, you'll notice the difference - players fall too fast, and as a result bounce too high.

ishitatsuyuki commented on src/pocketmine/entity/Entity.php:
```diff
@@ -1063,11 +1064,10 @@ public function fall($fallDistance){
-			}
+		
+		//Get the block directly beneath the player's feet, check if it is a slime block
+		if($this->getLevel()->getBlock($this->floor()->subtract(0, 1, 0)) instanceof SlimeBlock){
```
I prefer getSide.

ishitatsuyuki commented:
I think you should add a up velocity so the server will calculate it correctly.

dktapps commented:
@ishitatsuyuki I can't use getSide here, that code gets a block relative to the player's position as opposed to on the face of another block. And it's not the up velocity that's the problem; it's that the down velocity is mis-calculated. I'm not sure how to fix that.

ishitatsuyuki commented:
This is approved. The bug you mentioned may be fixed by someone else as long as the comment is there.

[![Approved with PullApprove](https://img.shields.io/badge/pullapprove-approved-brightgreen.svg)](https://pullapprove.com/iTXTech/Genisys/pull-request/1088/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=iTXTech/Genisys)

ishitatsuyuki commented:
Oh, wait: please use rebase instead of merge.

ishitatsuyuki commented:
Here's a approve again, also a friendly ping: @iTXTech/developers

dktapps commented:
I've rebased the pull request. That merge commit was an accident. (Still learning about Git)

gaoyichuan commented:
Approved

ishitatsuyuki commented:
+1, merge
